src/backend/cdb/cdbtmutils.c

```diff
@@ -140,6 +140,44 @@ DtxProtocolCommandToString(DtxProtocolCommand command)
 	return "Unknown";
 }
 
+CommandTag
+DtxProtocolCommandToCmdtag(DtxProtocolCommand command)
+{
+	switch (command)
+	{
+		case DTX_PROTOCOL_COMMAND_NONE:
+			return CMDTAG_MPPDTX_COMMAND_NONE;
+		case DTX_PROTOCOL_COMMAND_ABORT_NO_PREPARED:
+			return CMDTAG_MPPDTX_COMMAND_ABORT_NO_PREPARED;
+		case DTX_PROTOCOL_COMMAND_PREPARE:
+			return CMDTAG_MPPDTX_COMMAND_PREPARE;
+		case DTX_PROTOCOL_COMMAND_ABORT_SOME_PREPARED:
+			return CMDTAG_MPPDTX_COMMAND_ABORT_SOME_PREPARED;
+		case DTX_PROTOCOL_COMMAND_COMMIT_ONEPHASE:
+			return CMDTAG_MPPDTX_COMMAND_COMMIT_ONEPHASE;
+		case DTX_PROTOCOL_COMMAND_COMMIT_PREPARED:
+			return CMDTAG_MPPDTX_COMMAND_COMMIT_PREPARED;
+		case DTX_PROTOCOL_COMMAND_ABORT_PREPARED:
+			return CMDTAG_MPPDTX_COMMAND_ABORT_PREPARED;
+		case DTX_PROTOCOL_COMMAND_RETRY_COMMIT_PREPARED:
+			return CMDTAG_MPPDTX_COMMAND_RETRY_COMMIT_PREPARED;
+		case DTX_PROTOCOL_COMMAND_RETRY_ABORT_PREPARED:
+			return CMDTAG_MPPDTX_COMMAND_RETRY_ABORT_PREPARED;
+		case DTX_PROTOCOL_COMMAND_RECOVERY_COMMIT_PREPARED:
+			return CMDTAG_MPPDTX_COMMAND_RECOVERY_COMMIT_PREPARED;
+		case DTX_PROTOCOL_COMMAND_RECOVERY_ABORT_PREPARED:
+			return CMDTAG_MPPDTX_COMMAND_RECOVERY_ABORT_PREPARED;
+		case DTX_PROTOCOL_COMMAND_SUBTRANSACTION_BEGIN_INTERNAL:
+			return CMDTAG_MPPDTX_COMMAND_SUBTRANSACTION_BEGIN_INTERNAL;
+		case DTX_PROTOCOL_COMMAND_SUBTRANSACTION_RELEASE_INTERNAL:
+			return CMDTAG_MPPDTX_COMMAND_SUBTRANSACTION_RELEASE_INTERNAL;
+		case DTX_PROTOCOL_COMMAND_SUBTRANSACTION_ROLLBACK_INTERNAL:
+			return CMDTAG_MPPDTX_COMMAND_SUBTRANSACTION_ROLLBACK_INTERNAL;
+	}
```

Comment:
Maybe change lines to save original order like

```suggestion
		case DTX_PROTOCOL_COMMAND_SUBTRANSACTION_ROLLBACK_INTERNAL:
			return CMDTAG_MPPDTX_COMMAND_SUBTRANSACTION_ROLLBACK_INTERNAL;
		case DTX_PROTOCOL_COMMAND_SUBTRANSACTION_RELEASE_INTERNAL:
			return CMDTAG_MPPDTX_COMMAND_SUBTRANSACTION_RELEASE_INTERNAL;
	}
```

https://github.com/arenadata/gpdb/blob/20fd1f7fb0069fbebecf280887b440611c6dcfcf/src/include/cdb/cdbtm.h#L113-L114

Reply:
reordered